File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Chapter 22/CS/WebServiceCallScript/App_Code/TimeService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Services;



    /// <summary>
    /// Summary description for TimeService
    /// </summary>
    [WebService(Namespace="TimeService")]
    [WebServiceBinding(ConformsTo = WsiProfiles.BasicProfile1_1)]
    // To allow this Web Service to be called from script, using ASP.NET AJAX, uncomment the following line. 
    [System.Web.Script.Services.ScriptService()]
    public class TimeService : System.Web.Services.WebService
    {

        //public TimeService()
        //{

        //    //Uncomment the following line if using designed components 
        //    //InitializeComponent(); 
        //}

        //[WebMethod]
        //public string HelloWorld() {
        //    return "Hello World";
        //}
        [WebMethod]
        public string ServerTime(string myName)
        {
            if (myName == "")
                throw new Exception("Please Enter Your Name");
            return string.Format("Hello {0}, the time on the server is:  {1}", myName, DateTime.Now.ToString());
        }


    }

File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Chapter 22/VB/WebServiceCallScriptVB/App_Code/TimeService.vb
Imports System.Web
Imports System.Web.Services
Imports System.Web.Services.Protocols

' To allow this Web Service to be called from script, using ASP.NET AJAX, uncomment the following line.
<System.Web.Script.Services.ScriptService()> _
<WebService(Namespace:="TimeService")> _
<WebServiceBinding(ConformsTo:=WsiProfiles.BasicProfile1_1)> _
<Global.Microsoft.VisualBasic.CompilerServices.DesignerGenerated()> _
Public Class TimeService
    Inherits System.Web.Services.WebService

    '<WebMethod()> _
    'Public Function HelloWorld() As String
    '    Return "Hello World"
    'End Function
    <WebMethod()> _
    Public Function ServerTime(ByVal myName As String) As String
        If myName = "" Then
            Throw New Exception("Please Enter Your Name")
        End If
        Return String.Format("Hello {0}, the time on the server is:  {1}", myName, DateTime.Now.ToString())
    End Function

End Class